core/templates/pages/exploration-editor-page/modal-templates/exploration-metadata-diff-modal.component.spec.ts

```diff
@@ -0,0 +1,142 @@
+  let newExplorationMetadataBackendDict: ExplorationMetadataBackendDict;
+  let oldExplorationMetadataBackendDict: ExplorationMetadataBackendDict;
+
+  let headers = {
+    leftPane: 'header 1',
+    rightPane: 'header 2'
+  };
+  let newMetadata: ExplorationMetadata | null;
+  let oldMetadata: ExplorationMetadata | null;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule
+      ],
+      declarations: [
+        ExplorationMetadataDiffModalComponent
+      ],
+      providers: [
+        NgbActiveModal,
+        HistoryTabYamlConversionService
+      ],
+      schemas: [
+        NO_ERRORS_SCHEMA
+      ],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ExplorationMetadataDiffModalComponent);
+    component = fixture.componentInstance;
+    explorationMetadataObjectFactory = TestBed.inject(
+      ExplorationMetadataObjectFactory);
+    historyTabYamlConversionService = TestBed.inject(
+      HistoryTabYamlConversionService);
+  });
+
+  beforeEach(() => {
+    oldExplorationMetadataBackendDict = {
+      title: '',
+      category: '',
+      objective: '',
+      language_code: 'en',
+      tags: [],
+      blurb: '',
+      author_notes: '',
+      states_schema_version: 50,
+      init_state_name: 'Introduction',
+      param_specs: {},
+      param_changes: [],
+      auto_tts_enabled: false,
+      correctness_feedback_enabled: true,
+      edits_allowed: true
+    };
+    newExplorationMetadataBackendDict = {
+      title: 'Exploration',
+      category: 'Algebra',
+      objective: 'To learn',
+      language_code: 'en',
+      tags: [],
+      blurb: '',
+      author_notes: '',
+      states_schema_version: 50,
+      init_state_name: 'Introduction',
+      param_specs: {},
+      param_changes: [],
+      auto_tts_enabled: false,
+      correctness_feedback_enabled: true,
+      edits_allowed: true
+    };
+
+    newMetadata = explorationMetadataObjectFactory
+      .createFromBackendDict(newExplorationMetadataBackendDict);
+    oldMetadata = explorationMetadataObjectFactory
+      .createFromBackendDict(oldExplorationMetadataBackendDict);
+
+    component.headers = headers;
+    component.newMetadata = newMetadata;
+    component.oldMetadata = oldMetadata;
+  });
+
+  it('should initialize component properties after component is initialized',
+    fakeAsync(() => {
+      spyOn(
+        historyTabYamlConversionService, 'getYamlStringFromStateOrMetadata'
+      ).and.resolveTo('Yaml data');
+
+      component.ngOnInit();
+      tick(201);
+
+      expect(component.headers).toEqual(headers);
+      expect(component.newMetadata).not.toBeNull();
+      expect(component.oldMetadata).not.toBeNull();
+      expect(component.newMetadata?.toBackendDict()).toEqual(
+        newExplorationMetadataBackendDict);
+      expect(component.oldMetadata?.toBackendDict()).toEqual(
+        oldExplorationMetadataBackendDict);
```

Comment:
Should there be a pre-check for these?

Reply:
I just noticed that component.ngOnInit has no role in initializing these properties and these are supplied by the caller of this modal component. Hence, we don't need this test anymore.